broker-core/src/test/java/io/zeebe/broker/workflow/gateway/EventbasedGatewayTest.java

```diff
@@ -0,0 +1,196 @@
+  public ClientApiRule apiRule = new ClientApiRule(brokerRule::getClientAddress);
+
+  @Rule public RuleChain ruleChain = RuleChain.outerRule(brokerRule).around(apiRule);
+
+  private PartitionTestClient testClient;
+
+  @Before
+  public void init() {
+    testClient = apiRule.partitionClient();
+  }
+
+  @Test
+  public void testLifecycle() {
+    // given
+    testClient.deploy(WORKFLOW_WITH_TIMERS);
+
+    testClient.createWorkflowInstance(PROCESS_ID);
+
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CREATED).limit(2).exists());
+
+    // when
+    brokerRule.getClock().addTime(Duration.ofSeconds(1));
+
+    // then
+    assertThat(
+            RecordingExporter.workflowInstanceRecords()
+                .skipUntil(
+                    r -> r.getMetadata().getIntent() == WorkflowInstanceIntent.GATEWAY_ACTIVATED)
+                .limitToWorkflowInstanceCompleted())
+        .extracting(r -> tuple(r.getValue().getElementId(), r.getMetadata().getIntent()))
+        .containsExactly(
+            tuple("gateway", WorkflowInstanceIntent.GATEWAY_ACTIVATED),
+            tuple("timer-1", WorkflowInstanceIntent.CATCH_EVENT_TRIGGERING),
+            tuple("timer-1", WorkflowInstanceIntent.CATCH_EVENT_TRIGGERED),
+            tuple("to-end1", WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN),
+            tuple("end1", WorkflowInstanceIntent.END_EVENT_OCCURRED),
+            tuple(PROCESS_ID, WorkflowInstanceIntent.ELEMENT_COMPLETING),
+            tuple(PROCESS_ID, WorkflowInstanceIntent.ELEMENT_COMPLETED));
+  }
+
+  @Test
+  public void shouldCreateTimers() {
+    // given
+    testClient.deploy(WORKFLOW_WITH_TIMERS);
+
+    // when
+    testClient.createWorkflowInstance(PROCESS_ID);
+
+    // then
+    final Record<WorkflowInstanceRecordValue> gatewayEvent =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.GATEWAY_ACTIVATED)
+            .getFirst();
+
+    final List<Record<TimerRecordValue>> timerEvents =
+        RecordingExporter.timerRecords(TimerIntent.CREATED).limit(2).asList();
+
+    assertThat(timerEvents)
+        .hasSize(2)
+        .extracting(
+            r -> tuple(r.getValue().getHandlerFlowNodeId(), r.getValue().getElementInstanceKey()))
+        .contains(tuple("timer-1", gatewayEvent.getKey()), tuple("timer-2", gatewayEvent.getKey()));
+  }
+
+  @Test
+  public void shouldTriggerFirstEvent() {
+    // given
+    testClient.deploy(WORKFLOW_WITH_TIMERS);
+
+    testClient.createWorkflowInstance(PROCESS_ID);
+
+    final Record<WorkflowInstanceRecordValue> gatewayEvent =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.GATEWAY_ACTIVATED)
+            .getFirst();
+
+    // when
+    brokerRule.getClock().addTime(Duration.ofSeconds(1));
+
+    // then
+    final Record<TimerRecordValue> triggeredEvent =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED).getFirst();
+    Assertions.assertThat(triggeredEvent.getValue())
```

Comment:
Unnecessary `Assertions.`

Reply:
We have our own `Assertions` class with `assertThat` as well? That's...confusing if you don't know about it haha :see_no_evil: